src/graphql/dataSource.js

```diff
@@ -0,0 +1,169 @@
+const {DataSource} = require('apollo-datasource');
+const {search} = require('../operations/search/search');
+const {getRequestInfo} = require('./requestInfoHelper');
+const {searchById} = require('../operations/searchById/searchById');
+const {logWarn} = require('../operations/common/logging');
+const async = require('async');
+
+// const DataLoader = require('dataloader');
+
+class FhirDataSource extends DataSource {
+    constructor() {
+        super();
+        // this.dataLoader = new DataLoader(keys => myBatchGetUsers(keys));
+        this.meta = [];
+    }
+
+    initialize(config) {
+
+        return super.initialize(config);
+    }
+
+    /**
+     * This functions takes a FHIR Bundle and returns the resources in it
+     * @param {{entry:{resource: Resource}[]}} bundle
+     * @return {Resource[]}
+     */
+    unBundle(bundle) {
+        if (bundle.meta) {
+            this.meta.push(bundle.meta);
+        }
+        return bundle.entry.map(e => e.resource);
+    }
+
+    /**
+     * This is to handle unions in GraphQL
+     * @param obj
+     * @param context
+     * @param info
+     * @return {null|string}
+     */
+    // noinspection JSUnusedLocalSymbols
+    // eslint-disable-next-line no-unused-vars
+    resolveType(obj, context, info) {
+        if (Array.isArray(obj)) {
```

Comment:
Could we invert if to be 'if (!Array.isArray(obj)) return obj.resourceType; This could help reduce if nesting